src/transformers/models/bark/modeling_bark.py

```diff
@@ -0,0 +1,1504 @@
+
+            # transfer over info into model_in and convert to numpy
+            for n_inner in range(n_coarse, self.config.n_fine_codebooks):
+                fine_input[
+                    :, start_fill_idx : start_fill_idx + (max_fine_input_length - rel_start_fill_idx), n_inner
+                ] = input_buffer[:, rel_start_fill_idx:, n_inner]
+            del input_buffer
+
+        fine_input = fine_input.transpose(1, 2)[:, :, n_history:]
+        if n_remove_from_end > 0:
+            fine_input = fine_input[:, :, :-n_remove_from_end]
+
+        if fine_input.shape[-1] != coarse_output.shape[-2]:
+            raise ValueError("input and output should have the same seq_len")
+
+        return fine_input
+
+    def codec_decode(self, fine_output):
+        """Turn quantized audio codes into audio array using encodec."""
+
+        fine_output = fine_output.transpose(0, 1)
+        emb = self.codec_model.quantizer.decode(fine_output)
+        out = self.codec_model.decoder(emb)
+        audio_arr = out.squeeze(1)  # squeeze the codebook dimension
+
+        # TODO: del or no del?
+        # del fine_output, emb, out
+
+        return audio_arr
+
+    # @torch.no_grad
+    # _inference_mode()
+
+    def generate_audio(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        history_prompt: Optional[Dict[str, torch.Tensor]] = None,
+        **kwargs,
+    ) -> torch.LongTensor:
+        """
+        Generates audio from an input prompt and an additional optional `Bark` speaker prompt.
+
+        Args:
+            input_ids (Optional[torch.Tensor] of shape (batch_size, seq_len), optional):
+                Input ids. Will be truncated up to 256 tokens. Note that the output audios will be as long as the
+                longest generation among the batch.
+            The last token is `semantic_infer_token`. Note that batch_size is set to 1 to generate one audio per audio. Defaults to None.:
+            history_prompt (Optional[Dict[str,torch.Tensor]], optional):
+                Optional `Bark` speaker prompt. Defaults to None. Note that for now, this model takes only one speaker
+                prompt per batch.
+            max_coarse_history (int, optional):
+                Max length of the output of the coarse acoustics model used in the fine generation step. Defaults to
+                630.
+            sliding_window_len (int, optional):
+                The coarse generation step uses a sliding window to generate raw audio. Defaults to 60.
+        Returns:
+            torch.LongTensor: Output generated audio.
+
+        Example:
+
+        ```python
+        >>> from transformers import AutoProcessor, BarkModel
+        >>> from scipy.io.wavfile import write as write_wav
+
+
+        >>> processor = AutoProcessor.from_pretrained("ylacombe/bark-small")
+        >>> model = BarkModel.from_pretrained("ylacombe/bark-small")
+
+        >>> inputs, _ = processor("Hello, my dog is cute, I need him in my life")
```

Comment:
Hmmm this is a bit of a non-intuitive API - let's only return one output from the `processor`